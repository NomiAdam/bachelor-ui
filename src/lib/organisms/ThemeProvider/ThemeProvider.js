import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as Provider } from 'styled-components';

class ThemeProvider extends Component {
    shouldComponentUpdate(prevProps) {
        const { theme } = this.props;
        return JSON.stringify(prevProps.theme) !== JSON.stringify(theme);
    }

    render() {
        const { theme, children } = this.props;
        return (
            <Provider theme={ theme }>
                {children}
            </Provider>
        );
    }
}

ThemeProvider.propTypes = {
    children: PropTypes.any,
    theme: PropTypes.object,
};

export default ThemeProvider;
