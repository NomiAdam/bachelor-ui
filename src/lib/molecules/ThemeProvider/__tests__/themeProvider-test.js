import { shallow } from 'enzyme/build';
import React from 'react';
import ThemeProvider from '../ThemeProvider';

describe('ThemeProvider', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });
    it('should be defined', () => {
        expect(shallow(<ThemeProvider />)).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<ThemeProvider />)).toMatchSnapshot();
    });
});
