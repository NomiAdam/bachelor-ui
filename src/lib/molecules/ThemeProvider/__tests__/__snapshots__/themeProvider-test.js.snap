// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider should match the snapshot 1`] = `<ThemeProvider />`;
